app/controllers/ops_controller/settings/upload.rb

```diff
@@ -115,4 +115,22 @@ def logo_dir
     Dir.mkdir(dir) unless dir.exist?
     dir.to_s
   end
+
+  def valid_file?(file, type)
+    if type == 'ico'
+      valid_magic_number = "00000100"
+      magic_number_length = 4
+    else
+      valid_magic_number = "89504e470d0a1a0a" # .png format
+      magic_number_length = 8
+    end
+
+    temp_filepath = file.tempfile.path
+    tempfile = File.new(temp_filepath, 'r')
+    tempfile.rewind
+    magic_number = tempfile.readpartial(magic_number_length).unpack("H*").first
+    file_extensions = file.original_filename.split(".")
+
+    file_extensions.last.downcase == type && magic_number.downcase == valid_magic_number
```

Comment:
I think we can simplify this a little bit

1. Use a case statement and bail on unexpected tyes
2. Use the magic number directly (easier to read and verify in code)
3. Use binary read mode (more accurately compares bytes)

```suggestion
    case type
    when "ico"
      valid_magic_number = "\x00\x00\x01\x00".force_encoding("ASCII-8BIT")
    when "png"
      valid_magic_number = "\x89PNG\r\n\x1A\n".force_encoding("ASCII-8BIT")
    else
      return false
    end

    temp_filepath = file.tempfile.path
    tempfile = File.new(temp_filepath, 'rb')
    tempfile.rewind
    magic_number = tempfile.readpartial(valid_magic_number.size)
    file_extensions = file.original_filename.split(".")

    file_extensions.last.downcase == type && magic_number== valid_magic_number
```